lit_gpt/adapter_v2.py

```diff
@@ -123,7 +123,7 @@ def __init__(self, config: Config, block_idx: int) -> None:
         # key, query, value projections for all heads, but in a batch
         self.attn = AdapterV2Linear(in_features=config.n_embd, out_features=shape, bias=config.bias)
         # output projection
-        self.proj = AdapterV2Linear(config.n_embd, config.n_embd, bias=config.bias)
+        self.proj = AdapterV2Linear(config.head_size * config.n_head, config.n_embd, bias=config.bias)
```

Comment:
Can you add a comment explaining why this can't be `n_embd`?

Reply:
This is an interesting one actually. It makes sense because for the projection layer, you can have an arbitrary head dimension and number of heads as input but all LLMs I know use `head_dim = n_embd/n_head`. Actually, this is even hard-coded in [`torch.nn.MultiheadAttention`](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html). Imho, the way you implement it should be what everyone should implement to keep the code more general. So yeah, In short, I would definitely add a comment here :)